src/Microsoft.AspNet.Mvc.Core/ControllerActionDescriptorBuilder.cs

```diff
@@ -299,6 +300,22 @@ private static ParameterDescriptor CreateParameterDescriptor(ParameterModel para
             }
         }
 
+        private static void AddProperties(
+            ControllerActionDescriptor actionDescriptor,
+            ActionModel action,
+            ControllerModel controller)
+        {
+            foreach (var key in controller.Properties.Keys)
```

Comment:
You could iterate over the dictionary rather than the keys in the dictionary to avoid the extra lookup from the source